x-pack/plugins/apm/public/components/shared/Stacktrace/FrameHeading.tsx

```diff
@@ -33,16 +34,43 @@ const FrameHeading: React.SFC<Props> = ({ stackframe, isLibraryFrame }) => {
     ? LibraryFrameFileDetail
     : AppFrameFileDetail;
   const lineNumber: number = get(stackframe, 'line.number');
+  const messageWithLineNumber = (
+    <FormattedMessage
+      id="xpack.apm.stacktraceTab.exceptionLocationWithLineNumberMessage"
+      defaultMessage="{fileName} in {functionName} at {lineNumber}"
+      values={{
+        fileName: <FileDetail>{stackframe.filename}</FileDetail>,
+        functionName: <FileDetail>{stackframe.function}</FileDetail>,
+        lineNumber: (
+          <FileDetail>
+            <FormattedMessage
+              id="xpack.apm.stacktraceTab.exceptionLocationWithLineNumberMessage.lineNumberText"
+              defaultMessage="line {stackframeLineNumber}"
+              values={{
+                stackframeLineNumber: stackframe.line.number
+              }}
+              description="Part of composite text: xpack.apm.stacktraceTab.exceptionLocationWithLineNumberMessage
+                  + xpack.apm.stacktraceTab.exceptionLocationWithLineNumberMessage.lineNumberText"
+            />
+          </FileDetail>
+        )
+      }}
+    />
+  );
+  const messageWithoutLineNumber = (
+    <FormattedMessage
+      id="xpack.apm.stacktraceTab.exceptionLocationMessage"
+      defaultMessage="{fileName} in {functionName}"
+      values={{
+        fileName: <FileDetail>{stackframe.filename}</FileDetail>,
+        functionName: <FileDetail>{stackframe.function}</FileDetail>
+      }}
+    />
+  );
+
   return (
     <FileDetails>
-      <FileDetail>{stackframe.filename}</FileDetail> in{' '}
-      <FileDetail>{stackframe.function}</FileDetail>
-      {lineNumber > 0 && (
-        <Fragment>
-          {' at '}
-          <FileDetail>line {stackframe.line.number}</FileDetail>
-        </Fragment>
-      )}
+      {lineNumber > 0 ? messageWithLineNumber : messageWithoutLineNumber}
```

Comment:
This is my biggest gripe with the internationalisation approach: 9 simple lines turn into 37 nested, more complex lines. I don't have any good suggestions to change this, but it definitely affects readability of the code negatively.

Reply:
Btw. sorry for my unconstructive comment. I very much appreciate the help we are getting to localize APM!

Reply:
As an option we can use `select` like in the example (this is similar to the previous version with `plural`):

```js
const FrameHeading: React.SFC<Props> = ({ stackframe, isLibraryFrame }) => {
  const FileDetail = isLibraryFrame
    ? LibraryFrameFileDetail
    : AppFrameFileDetail;
  const lineNumber: number = get(stackframe, 'line.number');
  return (
    <FileDetails>
      <FormattedMessage
        id="xpack.apm.stacktraceTab.exceptionLocationMessage"
        defaultMessage="{fileName} in {functionName} {lineNumber, select, 0 {} other {at {lineNumberDescription}}}"
        values={{
          fileName: <FileDetail>{stackframe.filename}</FileDetail>,
          functionName: <FileDetail>{stackframe.function}</FileDetail>,
          lineNumber,
          lineNumberDescription: (
            <FileDetail>
              <FormattedMessage
                id="xpack.apm.stacktraceTab.exceptionLocationMessage.lineNumberText"
                defaultMessage="line {stackframeLineNumber}"
                values={{
                  stackframeLineNumber: stackframe.line.number
                }}
                description="Part of composite text: xpack.apm.stacktraceTab.exceptionLocationMessage
                  + xpack.apm.stacktraceTab.exceptionLocationMessage.lineNumberText"
              />
            </FileDetail>
          )
        }}
      />
    </FileDetails>
  );
};
```

We also can agreed to not translate this label since it's like a piece of a stacktrace

![image](https://user-images.githubusercontent.com/11524637/50840365-66e59700-1373-11e9-8f2d-86446bd71b98.png)

Reply:
> We also can agreed to not translate this label since it's like a piece of a stacktrace

If you think it's ok I'm also good with that 👍

Reply:
I think that in this case it's worth to leave the label not-translated.
Done.